### PR TITLE
[FEATURE] Setup initial frontend codebase for GCA - LRN-34258

### PR DESCRIPTION
**Pull request by christian-crisologo-lrn**
Initial setup for GCA frontend codebase. 
This is just a frontend-only for the style guide placeholder

- [x] NPM + Yarn
- [x] Webpack with dev + prod build
- [x] SCSS support
- [x] ReactJs + typescript
- [x] Jest test

TODO: 

- [ ] Github workflow to run the build when the branch merged or when the PR closed

### CODE REVIEW

File: www/assessment/customization/style-guide/.npmrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+//npm.pkg.github.com/:_authToken=${NPM_TOKEN}
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/.npmrc:1]: We do not use anything from LRN private packages here so remove this file

File: www/assessment/customization/style-guide/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    "plugin:react/recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "ignorePatterns": [],
+  "plugins": [
+    "@typescript-eslint",
+    "react-hooks"
+  ],
+  "globals": {
+    "__LRN_DEVELOPMENT": true,
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/.eslintrc:14]: This is not necessary as you are not building a proper client facing API that needs to differentiate bw DEV vs PROD

File: www/assessment/customization/style-guide/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const path = require('path');
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/jest.config.js:1]: Remove this file, I am not sure we will even have unit-tests in this first phase

File: www/assessment/customization/style-guide/tests/specs/@testConfig/jest.setup.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Any setup that we need to do before executing all the tests
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/tests/specs/@testConfig/jest.setup.js:1]: Same here

File: www/assessment/customization/style-guide/tests/specs/@testConfig/styleMock.js [hunk 5]
```diff
@@ -0,0 +1 @@
+module.exports = {};
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/tests/specs/@testConfig/styleMock.js:1]: And here

File: www/assessment/customization/style-guide/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+    "compilerOptions": {
+        "moduleResolution": "node",
+        "baseUrl": "src",
+        "paths": {
+            "*": ["*"],
+            "@testConfig/*": ["tests/specs/@testConfig/*"]
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/tsconfig.json:7]: Delete this

File: www/assessment/customization/style-guide/webpack.config.base.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
+/* globals require, __dirname */
+const path = require('path');
+const webpack = require('webpack');
+const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
+const autoprefixer = require('autoprefixer');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/webpack.config.base.js:6]: Remove all this, if you are going to use predefined Libraries for UI like Material UI, keep this set up as small as possible

File: www/assessment/customization/style-guide/webpack.config.base.js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+                configOverwrite: {
+                    exclude: [
+                        'node_modules',
+                        '**/*.spec.ts',
+                        '**/*.spec.tsx',
+                    ],
+                },
+            },
+        }),
+        new MiniCssExtractPlugin({
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/webpack.config.base.js:52]: Strip this out

File: www/assessment/customization/style-guide/webpack.config.base.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
+/* globals require, __dirname */
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/webpack.config.base.js:1]: Rename this file to `webpack.config.js` only

File: www/assessment/customization/style-guide/webpack.config.prod.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+/*global require */
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/webpack.config.prod.js:1]: Remove this file

File: www/assessment/customization/style-guide/webpack.config.dev.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+const webpack = require('webpack');
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/webpack.config.dev.js:1]: Remove this file

File: www/assessment/customization/style-guide/README.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
+##Style guide custimization for Assessment UI
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/README.md:1]: Update this README

File: www/assessment/customization/style-guide/webpack.config.base.js [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+                },
+            },
+        },
+    },
+    module: {
+        rules: [
+            {
+                test: /\.scss$/,
+                use: [
+                    MiniCssExtractPlugin.loader,
```
Comment (jack-vo) [inline www/assessment/customization/style-guide/webpack.config.base.js:84]: Remove this line

**Review by jack-vo — CHANGES_REQUESTED** [review]

**Review by eric-guo-lrn — APPROVED** [review]
LGTM

**Pull request merged**